src/ssz/type/container.zig

```diff
@@ -333,6 +346,20 @@ pub fn VariableContainerType(comptime ST: type) type {
             try merkleize(@ptrCast(&chunks), chunk_depth, out);
         }
 
+        /// Creates a new `VariableContainerType` and clones all underlying fields in the container.
+        ///
+        /// Caller owns the memory.
+        pub fn clone(
+            allocator: std.mem.Allocator,
+            value: *const Type,
+            out: *Type,
+        ) !void {
+            inline for (fields) |field| {
+                @field(out, field.name) = field.type.default_value;
+                try field.type.clone(allocator, &@field(value, field.name), &@field(out, field.name));
```

Comment:
```suggestion
                @field(out, field.name) = field.type.default_value;
                if (comptime isFixedType(field.type) {
                    try field.type.clone(&@field(value, field.name), &@field(out, field.name));
                } else {
                    try field.type.clone(allocator, &@field(value, field.name), &@field(out, field.name));
                }
```

Reply:
Addressed in [88f2fb0](https://github.com/ChainSafe/ssz-z/pull/51/commits/88f2fb0366abfeaaeb296bc736982d700f89feed)